src/apps/search-result/search-result.tsx

```diff
@@ -168,22 +158,23 @@ const SearchResult: React.FC<SearchResultProps> = ({ q, pageSize }) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [campaignData]);
 
+  // This is a hack to clear filters when the user navigates
+  // to the search page from links where there should be no filters.
+  useEffect(() => {
+    const filtersUrlParam = getUrlQueryParam("filters");
+    if (filtersUrlParam !== "usePersistedFilters") clearFilter();
+  }, [clearFilter]);
```

Comment:
I had to think a bit about why this is necessary.

Letting the user navigate to a new search result should not necessarily clear the filters for any other search result. However we only have a single place to store the selected filters so if the user adds more filters on the new search result it may result in filters getting mixed up between searches.

Consequently we have to clear them when starting a new search. This will however result in the following situation:

1. The user does a search
2. The user applies a filter
3. The user goes to a material
4. The user clicks on a term in a material
5. The user is shown a search result
6. Filters are cleared
7. The user clicks the back button twice and returns to the search result in 1
8. Filters from 2 are not applied as they have been cleared in 6

😵